src/allmydata/immutable/encode.py

```diff
@@ -694,3 +695,20 @@ def get_uri_extension_data(self):
         return self.uri_extension_data
     def get_uri_extension_hash(self):
         return self.uri_extension_hash
+
+    def get_uri_extension_size(self):
+        """
+        Calculate the size of the URI extension that gets written at the end of
+        immutables.
+
+        This may be done earlier than actual encoding, so e.g. we might not
+        know the crypttext hashes, but that's fine for our purposes since we
+        only care about the length.
+        """
+        params = self.uri_extension_data.copy()
+        assert params
+        params["crypttext_hash"] = b"\x00" * 32
```

Comment:
`allmydata.util.hashutil.CRYPTO_VAL_SIZE` might be a better spelling than `32` as far as redundancy/divergence goes.